lte/gateway/c/core/oai/tasks/sgw/sgw_context_manager.cpp

```diff
@@ -318,35 +305,44 @@ sgw_eps_bearer_ctxt_t* sgw_cm_get_eps_bearer_entry(
 
 s_plus_p_gw_eps_bearer_context_information_t* sgw_cm_get_spgw_context(
     teid_t teid) {
-  s_plus_p_gw_eps_bearer_context_information_t* spgw_bearer_context_info = NULL;
-  hash_table_ts_t* state_teid_ht = get_spgw_teid_state();
+  s_plus_p_gw_eps_bearer_context_information_t* spgw_bearer_context_info =
+      nullptr;
+  state_teid_map_t* state_teid_map = get_spgw_teid_state();
+  if (!state_teid_map) {
+    OAILOG_ERROR(LOG_SPGW_APP, "Failed to get state_teid_map");
+    return spgw_bearer_context_info;
+  }
 
-  hashtable_ts_get(state_teid_ht, (const hash_key_t)teid,
-                   (void**)&spgw_bearer_context_info);
+  state_teid_map->get(teid, &spgw_bearer_context_info);
   return spgw_bearer_context_info;
 }
 
 spgw_ue_context_t* spgw_get_ue_context(imsi64_t imsi64) {
   OAILOG_FUNC_IN(LOG_SPGW_APP);
   spgw_ue_context_t* ue_context_p = NULL;
-  hash_table_ts_t* state_ue_ht = get_spgw_ue_state();
-  hashtable_ts_get(state_ue_ht, (const hash_key_t)imsi64,
-                   (void**)&ue_context_p);
+  map_uint64_spgw_ue_context_t* state_ue_map = get_spgw_ue_state();
+  if (!state_ue_map) {
+    OAILOG_ERROR(LOG_SPGW_APP, "Failed to find spgw_ue_state");
+    OAILOG_FUNC_RETURN(LOG_SPGW_APP, ue_context_p);
```

Comment:
Same comment as above
```suggestion
    OAILOG_FUNC_RETURN(LOG_SPGW_APP, nullptr);
```